data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java

```diff
@@ -133,6 +133,7 @@ private Future<HttpResponse<Buffer>> send(final CloudEvent event, final int retr
           if (isRetryableStatusCode(response.statusCode()) && retryCounter < consumerVerticleContext.getEgressConfig().getRetry()) {
             return retry(retryCounter, event);
           }
+          return Future.failedFuture(cause);
```

Comment:
@pierDipi did we miss this, when we did some of the refactorings?

Need to port to upstream too?

Reply:
It's coming from upstream: https://github.com/knative-sandbox/eventing-kafka-broker/commit/3d84363ad294356f15717d6585e1f7b2ed403fec